Test failure in tests_high_five_base.cpp

I'm just trying to build HighFive in a Docker container (Ubuntu 20). Options are:
* -DHIGHFIVE_EXAMPLES=OFF 
* -DCMAKE_INSTALL_PREFIX=/usr/HighFive 
* -DHIGHFIVE_PARALLEL_HDF5=ON
Boost is installed. 

When make gets to "Building CXX object tests" the process falls over at the file `HighFive/tests/unit/tests_high_five_base.cpp`, due to the warning `'haddr_t HighFive::ObjectInfo::getAddress() const' is deprecated`. Plus, another two warning for thrown for `redunant move in return statement`.

It doesn't look like this is a quirk of the container, and equally it doesn't look like this is really a blocking error. Should I have set a different flag to skip some deprecated tests?

Happens on the latest commit, and on the 2.2.2 release. 2.2 release just gives the deprecation error and installs fine.

Full trace:
```
/HighFive/tests/unit/tests_high_five_base.cpp: In member function 'void DataTypeEqualTakeBack::test_method()':
/HighFive/tests/unit/tests_high_five_base.cpp:465:46: warning: 'haddr_t HighFive::ObjectInfo::getAddress() const' is deprecated: "Deprecated since HighFive 2.2. Soon supporting VOL tokens" [-Wdeprecated-declarations]
  465 |     auto addr = dataset.getInfo().getAddress();
      |                                              ^
In file included from /HighFive/include/highfive/H5Object.hpp:123,
                 from /HighFive/include/highfive/H5DataSpace.hpp:29,
                 from /HighFive/include/highfive/H5DataSet.hpp:14,
                 from /HighFive/tests/unit/tests_high_five_base.cpp:21:
/HighFive/include/highfive/bits/H5Object_misc.hpp:99:16: note: declared here
   99 | inline haddr_t ObjectInfo::getAddress() const noexcept {
      |                ^~~~~~~~~~
In file included from /HighFive/include/highfive/bits/../H5Reference.hpp:80,
                 from /HighFive/include/highfive/bits/H5Converter_misc.hpp:30,
                 from /HighFive/include/highfive/bits/H5Attribute_misc.hpp:25,
                 from /HighFive/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /HighFive/include/highfive/H5File.hpp:79,
                 from /HighFive/tests/unit/tests_high_five_base.cpp:23:
/HighFive/include/highfive/bits/../bits/H5Reference_misc.hpp: In instantiation of 'T HighFive::Reference::dereference(const HighFive::Object&) const [with T = HighFive::Group]':
/HighFive/tests/unit/tests_high_five_base.cpp:1618:9:   required from here
/HighFive/include/highfive/bits/../bits/H5Reference_misc.hpp:46:25: error: redundant move in return statement [-Werror=redundant-move]
   46 |     return std::move(obj);
      |                         ^
/HighFive/include/highfive/bits/../bits/H5Reference_misc.hpp:46:25: note: remove 'std::move' call
/HighFive/include/highfive/bits/../bits/H5Reference_misc.hpp: In instantiation of 'T HighFive::Reference::dereference(const HighFive::Object&) const [with T = HighFive::DataSet]':
/HighFive/tests/unit/tests_high_five_base.cpp:1619:57:   required from here
/HighFive/include/highfive/bits/../bits/H5Reference_misc.hpp:46:25: error: redundant move in return statement [-Werror=redundant-move]
/HighFive/include/highfive/bits/../bits/H5Reference_misc.hpp:46:25: note: remove 'std::move' call
cc1plus: all warnings being treated as errors
make[2]: *** [tests/unit/CMakeFiles/tests_high_five_base.dir/build.make:63: tests/unit/CMakeFiles/tests_high_five_base.dir/tests_high_five_base.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:202: tests/unit/CMakeFiles/tests_high_five_base.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

--- a/include/highfive/bits/H5Reference_misc.hpp
+++ b/include/highfive/bits/H5Reference_misc.hpp
@@ -43,7 +43,11 @@ inline T Reference::dereference(const Object& location) const {
         HDF5ErrMapper::ToException<ReferenceException>(
             "Trying to dereference the wrong type");
     }
+#if defined __GNUC__ && __GNUC__ < 9
     return std::move(obj);
+#else
+    return obj;
+#endif
 }
 
 inline Object Reference::get_ref(const Object& location) const {

